libraries/recorder.js

```diff
@@ -83,11 +87,26 @@ recorder.getAndGuaranteeOutputFilename = function(logName) {
     return outputFilename;
 };
 
+/**
+ * Persist current state to file, skipping if no differences recorded since last persist
+ * @return {Promise<string|null>} log file name
+ */
 recorder.persistToFile = function () {
+    // recorder.objectOld is the latest state persisted in timeObject
+    // Because we're using objectOld as the basis for each log file, having
+    // only one entry in timeObject means `object` never had a difference from
+    // `objectOld` and we can skip writing to the file.
+    if (Object.keys(recorder.timeObject).length <= 1 && hasPersistedToFile) {
+        return Promise.resolve(null);
+    }
+
+    hasPersistedToFile = true;
```

Comment:
This is set just once for the first time after start. Is the intention to skip creation of new log file when there was no change at all since the last call of persistToFile()? Essentially, delaying the creation of next log file to some future call of persistToFile when there are some new changes to store.

Reply:
Yes, that's the goal with this overall pr